src/main.c

```diff
@@ -7,56 +7,40 @@
 #include "mmu.h"
 #include "cli.h"
 #include "logging.h"
-#include "errcodes.h"
 
 int main(const int argc, char **const argv) {
-    CLIArguments* args = cli_config_default();
-    
-    if (cli_config_handle(args, argc, argv)) {
-        cli_config_destroy(args);
-        exit(ERR_FAILURE);
-    }
+	CLIArguments cli_args;
+	cli_parse(&cli_args, argc, argv);
 
-    ErrorCode ret = rom_load("./roms/hello-world.gb");
-    if (ret) {
-        LOG_FATAL("Failed to load ROM");
-        exit(ret);
-    }
-    printf("Successfully initialized ROM\n");
+	log_set_lvl(cli_args.logging_level);
 
-    ret = mem_init();
-    if (ret) {
-        fprintf(stderr, "Failed to initialize MMU\n");
-        exit(ret);
-    }
+	rom_init(cli_args.rom_path);
+    LOG_INFO("Successfully initialized ROM");
+
+	mmu_init();
     LOG_INFO("Successfully initialized MMU");
 
-    ret = lcd_load();
-    if (ret) {
-        LOG_FATAL("Failed to initialize LCD");
-        exit(ret);
-    }
+	lcd_init();
     LOG_INFO("Successfully initialized LCD");
 
-
     cpu_init();
     LOG_INFO("Successfully initialized CPU");
 
     uint8_t iterations = 0;
     while (1) {
-        if (cpu_step())
-            break;
+        cpu_step();
```

Comment:
Couldn't CPU step "fail", should the CPU be halted? I know that this is not subject of this PR, but should be considered nonetheless.

Reply:
Absolutely, that should certainly be addressed somewhere 🤔 Should we leave it for now, or make a "reminder" in form of an issue?

Reply:
I'd open an issue for that

Reply:
Do you mind doing that?

Reply:
On second thought, there really is no way the CPU step should fail. Consulting the manual, there is no instruction to initiate shutdown. `HALT` and `STOP` are recoverable and thus should not terminate execution. I think leaving the function `void` should suffice for now. Should I have overlooked something, changes could be made quickly.

Reply:
Done, see #28.